localization/pose_estimator/ndt_scan_matcher/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -41,91 +42,102 @@ NDTScanMatcher::NDTScanMatcher(ros::NodeHandle nh, ros::NodeHandle private_nh)
+  ndt_pose_pub_ = this->create_publisher<geometry_msgs::msg::PoseStamped>("ndt_pose", 10);
   ndt_pose_with_covariance_pub_ =
-    nh_.advertise<geometry_msgs::PoseWithCovarianceStamped>("ndt_pose_with_covariance", 10);
+    this->create_publisher<geometry_msgs::msg::PoseWithCovarianceStamped>(
+      "ndt_pose_with_covariance", 10);
   initial_pose_with_covariance_pub_ =
-    nh_.advertise<geometry_msgs::PoseWithCovarianceStamped>("initial_pose_with_covariance", 10);
-  exe_time_pub_ = nh_.advertise<std_msgs::Float32>("exe_time_ms", 10);
-  transform_probability_pub_ = nh_.advertise<std_msgs::Float32>("transform_probability", 10);
-  iteration_num_pub_ = nh_.advertise<std_msgs::Float32>("iteration_num", 10);
+    this->create_publisher<geometry_msgs::msg::PoseWithCovarianceStamped>(
+      "initial_pose_with_covariance", 10);
+  exe_time_pub_ = this->create_publisher<autoware_debug_msgs::msg::Float32Stamped>("exe_time_ms", 10);
+  transform_probability_pub_ =
+    this->create_publisher<autoware_debug_msgs::msg::Float32Stamped>("transform_probability", 10);
+  iteration_num_pub_ = this->create_publisher<autoware_debug_msgs::msg::Int32Stamped>("iteration_num", 10);
   initial_to_result_distance_pub_ =
-    nh_.advertise<std_msgs::Float32>("initial_to_result_distance", 10);
+    this->create_publisher<autoware_debug_msgs::msg::Float32Stamped>("initial_to_result_distance", 10);
   initial_to_result_distance_old_pub_ =
-    nh_.advertise<std_msgs::Float32>("initial_to_result_distance_old", 10);
+    this->create_publisher<autoware_debug_msgs::msg::Float32Stamped>("initial_to_result_distance_old", 10);
   initial_to_result_distance_new_pub_ =
-    nh_.advertise<std_msgs::Float32>("initial_to_result_distance_new", 10);
-  ndt_marker_pub_ = nh_.advertise<visualization_msgs::MarkerArray>("ndt_marker", 10);
+    this->create_publisher<autoware_debug_msgs::msg::Float32Stamped>("initial_to_result_distance_new", 10);
+  ndt_marker_pub_ = this->create_publisher<visualization_msgs::msg::MarkerArray>("ndt_marker", 10);
   ndt_monte_carlo_initial_pose_marker_pub_ =
-    nh_.advertise<visualization_msgs::MarkerArray>("monte_carlo_initial_pose_marker", 10);
-  diagnostics_pub_ = nh_.advertise<diagnostic_msgs::DiagnosticArray>("/diagnostics", 10);
-
-  service_ = nh_.advertiseService("ndt_align_srv", &NDTScanMatcher::serviceNDTAlign, this);
+    this->create_publisher<visualization_msgs::msg::MarkerArray>(
+      "monte_carlo_initial_pose_marker", 10);
+  diagnostics_pub_ =
+    this->create_publisher<diagnostic_msgs::msg::DiagnosticArray>("/diagnostics", 10);
+
+  service_ = this->create_service<autoware_localization_srvs::srv::PoseWithCovarianceStamped>(
+    "ndt_align_srv",
+    std::bind(
+      &NDTScanMatcher::serviceNDTAlign, this, std::placeholders::_1, std::placeholders::_2));
   // setup dynamic reconfigure server
   // f_ = boost::bind(&NDTScanMatcher::configCallback, this, _1, _2);
   // server_.setCallback(f_);
```

Comment:
Yes, the configCallback doesn't seem to exist. Would you like to have this commented out code deleted?

Reply:
I'm not the original author so I don't know exactly. But to me since it cannot even be ported since the rest of the code doesn't exist, I don't really see the point in keeping it.